### ISSUE TITLE
Add form config `reviewErrors` docs

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#1615]
### Duplicate check

- [X] I've searched for any related issues and avoided creating a duplicate issue.

### This update is for:

Forms Library

### What is the name?

reviewErrors

### What is the nature of this update?

- [X] How to build this component/pattern
- [X] When to use this component/pattern
- [ ] When to use something else
- [ ] Usability guidance
- [X] Accessibility
- [X] Implementation
- [ ] Research insights
- [ ] Package information
- [X] Addition to Forms Library documentation
- [ ] Update to existing Forms Library documentation

### What problem does this solve?

When a user encounters a form error on the review & submit page, they are presented with a generic error message

The accordions are highlighted and an error is included below the header, but to improve accessibility, links that open the accordion and place the appropriate page into edit mode are desirable. This has been implemented for form 526 by adding two entries: `showReviewErrors` and `reviewErrors`.

The necessary **documentation has been included in the [`forms-system/docs`](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/docs/reviewErrors.md) folder**, but it not within the design system documentation and is therefore difficult to discover.

### Additional Context

This work has a [long history of review & evaluation](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2138).

**Comment by humancompanion-usds** [comment]
@Mottie - Are the images linked in the code docs up-to-date? Do we really put the Alert after the Accordions?

**Comment by Mottie** [comment]
Yes, the alert is after the accordions, and above the privacy statement.

I have work-in-progress review error links for our Supplemental Claims form (it has capitalization and content issues); but, we're going to need to update the review errors code, and thus the documentation. What we need to change is:

- Include data within the link text directly from the array it's targeting. For example, we want to replace "second private facility issue selection (select one or more issues)" with "Private hospital is missing a condition (select one or more conditions)"
- Point to a specific page in the flow, this allows us to take the user directly to the page with the error. We'll have to work out what happens after they update - go directly back to the review & submit page, or step back through all the remaining pages to return to the review & submit page. We may need to allow for both scenarios because altering data on an earlier page may change the form flow.

![Supplemental Claim review & submit page showing evidence accordion expanded with various errors, and an alert below the accordions with links that direct the user back to the specific chapter & page with the error (there are 14 error links)](https://user-images.githubusercontent.com/136959/234717224-e4548cd4-5318-413a-9f54-e35172ce6acb.png)

**Comment by humancompanion-usds** [comment]
If you are going to do this:

> Point to a specific page in the flow, this allows us to take the user directly to the page with the error. We'll have to work out what happens after they update - go directly back to the review & submit page, or step back through all the remaining pages to return to the review & submit page. We may need to allow for both scenarios because altering data on an earlier page may change the form flow.

then you are getting pretty close to the newer Review page design. We've tested it twice and it tested well. Do you all want to give that design a go? I can also draft some recommendations on how to handle the scenario you are describing.

**Comment by Mottie** [comment]
Yes! I'd love to check it out!

**Comment by humancompanion-usds** [comment]
Okay, the design is in two places:

* In Sketch it's on the "Review page" page and it is the v3 artboard. https://sketch.com/s/09640425-48be-4019-8277-8a924eff1df7
* In UXPin where we prototyped and tested it the first time and made some adjustments: https://preview.uxpin.com/3f788e8c6673bebc0365fd5d08a28347037aa809#/pages/150422644/simulate/sitemap?mode=i. (it is the "Burial form Review Page with section edit" variation)

The UXPin prototype is the more up-to-date version but, it got a few things wrong with the buttons for returning the user to the Review page or to the next step depending on what they changed (which is the scenario you described above). We put that onto the user in this prototype to figure out with multiple buttons. That was not what I intended and it failed miserably in testing. 

Ideally what we'd do is make the form "smart" and when the user clicks Continue do a check to determine if their changes have invalidated any of the data they previously entered. If something has been invalidated we'd take them to the most appropriate step in the process to remediate that with an alert informing them of what needs to be altered. If their changes have not invalidated anything then we'd return them to the Review page, anchoring to the section they had started editing, and reflecting their changes.
